modules/fedimint-lnv2-client/src/lib.rs

```diff
@@ -697,69 +733,44 @@ impl LightningClientModule {
     /// automatically.
     pub async fn receive_custom(
         &self,
-        invoice_amount: Amount,
-        expiry_time: u32,
+        amount: Amount,
+        expiry_secs: u32,
         description: Bolt11InvoiceDescription,
         payment_fee_limit: PaymentFee,
         gateway_api: Option<SafeUrl>,
-    ) -> Result<(Bolt11Invoice, OperationId), FetchInvoiceError> {
-        let (contract, .., invoice) = self
-            .create_contract_and_fetch_invoice_custom(
+        custom_meta: Value,
+    ) -> Result<(Bolt11Invoice, OperationId), ReceiveError> {
+        let (contract, invoice) = self
+            .create_contract_and_fetch_invoice(
                 self.keypair.public_key(),
-                invoice_amount,
-                expiry_time,
+                amount,
+                expiry_secs,
                 description,
                 payment_fee_limit,
                 gateway_api,
             )
             .await?;
 
         let operation_id = self
-            .receive_incoming_contract(contract)
+            .receive_incoming_contract(contract, invoice.clone(), custom_meta)
             .await
             .expect("The contract has been generated with our public key");
 
         Ok((invoice, operation_id))
     }
 
-    /// Create an incoming contract locked to a public key derived from the
-    /// recipient's static module public key and fetches the corresponding
-    /// invoice. To receive the payment the recipient needs to call
-    /// [`Self::receive_incoming_contract`] on the returned contract. For
-    /// testing you can optionally specify a gateway to generate the
-    /// invoice, otherwise a gateway will be selected automatically.
-    pub async fn create_contract_and_fetch_invoice(
+    // Create an incoming contract locked to a public key derived from the
+    // recipient's static module public key and fetches the corresponding
+    // invoice.
+    async fn create_contract_and_fetch_invoice(
```

Comment:
Why isn't it a doc comment anymore?

Reply:
Method is not public anymore.

Reply:
It's still useful to have it be a docstring since it will integrate better with tooling that can display docs (e.g. https://docs.fedimint.org/ renders private items). Not blocking, but for future PRs.